lib/grp.gi

```diff
@@ -5070,11 +5073,24 @@ end);
 #M  HasSolvableFactorGroup(<G>,<N>)   test whether G/N is solvable
 ##
 InstallGlobalFunction(HasSolvableFactorGroup,function(G,N)
-local s,l;
+local gen, D;
+
+  if HasIsSolvableGroup(G) and IsSolvableGroup(G) then
+    return true;
+  fi;
   Assert(2,IsNormal(G,N) and IsSubgroup(G,N));
-  s := DerivedSeriesOfGroup(G);
-  l := Length(s);
-  return IsSubgroup(N,s[l]);
+  D := G;
```

Comment:
Hmm, seems my earlier comment on this got lost? I suggested to check `HasDerivedSeriesOfGroup` here and if true, fallback to the previous code, which could be more efficient here.  Thinking about it now, though, I think it might be faster in some cases but could also be slower in others... hmmm